apps/web/src/modules/dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,178 @@
+  currentAuction?: CurrentAuctionFragment | null
+}
+
+const fetchDaoProposalState = async (dao: DashboardDao) => {
+  const proposals = await Promise.all(
+    dao.proposals.map(async (proposal) => {
+      const proposalState = await getProposalState(
+        dao.chainId,
+        proposal.dao.governorAddress,
+        proposal.proposalId
+      )
+      return { ...proposal, proposalState: proposalState }
+    })
+  )
+  return {
+    ...dao,
+    proposals: proposals.filter((proposal) =>
+      ACTIVE_PROPOSAL_STATES.includes(proposal.proposalState)
+    ),
+  }
+}
+
+const fetchDashboardData = async (address: string) => {
+  try {
+    const userDaos = (await dashboardRequest(address)) as unknown as DashboardDao[]
+    if (!userDaos) throw new Error('Dashboard DAO query returned undefined')
+    const resolved = await Promise.all(userDaos.map(fetchDaoProposalState))
+    return resolved
+  } catch (error) {
+    throw new Error('Error fetching dashboard data')
+  }
+}
+
+const Dashboard = () => {
+  const { address } = useAccount()
+
+  const { data, error, isValidating } = useSWR(
+    [`${SWR_KEYS.DASHBOARD}:${address}`],
+    address ? () => fetchDashboardData(address) : null,
+    { revalidateOnFocus: false }
+  )
+
+  const [mutating, setMutating] = useState(false)
+
+  if (error) {
+    return (
+      <DashPage>
+        <Text fontSize={18} mb={'x6'}>
+          Something went wrong.
+        </Text>
+        <DisplayPanel
+          title="Error"
+          description={
+            error?.message || 'Error fetching display data. Error message not found.'
+          }
+        />
+      </DashPage>
+    )
+  }
+  if (isValidating && !mutating) {
+    return (
+      <DashboardLayout
+        auctionCards={Array.from({ length: 3 }).map((_, i) => (
+          <AuctionCardSkeleton key={`auctionCardSkeleton:${i}`} />
+        ))}
+        daoProposals={
+          <Box>
+            <DAOCardSkeleton />
+            {Array.from({ length: 2 }).map((_, i) => (
+              <ProposalCardSkeleton key={`daoCardSkeleton:${i}`} />
+            ))}
+          </Box>
+        }
+      />
+    )
+  }
+  if (!address) {
+    return <DashConnect />
+  }
+  if (!data?.length) {
+    return (
+      <DashPage>
+        <Text fontSize={18}>It looks like you haven’t joined any DAOs yet.</Text>
+        <DaoFeed isDashboard />
+      </DashPage>
+    )
+  }
+
+  const handleMutate = async () => {
+    setMutating(true)
+    await mutate([`${SWR_KEYS.DASHBOARD}:${address}`], () => fetchDashboardData(address))
+    setMutating(false)
+  }
+
+  const hasLiveProposals = data.some((dao) => dao.proposals.length)
+
+  return (
+    <DashboardLayout
+      auctionCards={data.map((dao) => (
+        <DaoAuctionCard
+          key={`auctionCard:${dao.tokenAddress}`}
+          {...dao}
+          userAddress={address}
+          handleMutate={handleMutate}
+        />
+      ))}
+      daoProposals={
+        hasLiveProposals ? (
```

Comment:
can we make this a separate constant for readability and i think we should memoize this so that `<DaoAuctionCard />` isn't being unnecessarily  re-rendered